pkg/sources/github/github.go

```diff
@@ -793,36 +705,41 @@ func (s *Source) scan(ctx context.Context, installationClient *github.Client, ch
 			}
 
 			// Scan the repository
+			repoInfo, ok := s.repoInfoCache.get(repoURL)
+			if !ok {
+				// This should never happen.
+				err := fmt.Errorf("no repoInfo for URL: %s", repoURL)
+				s.log.Error(err, "failed to scan repository")
```

Comment:
This seems to get populated when using `--org` but not `--repo. Gotta look into that.

Reply:
Quickly looked at this and found:

* `conn.Repositories` is set via the `--repo` flag
* `Init` takes the repos from there, normalizes them, and saves it in the `filteredRepoCache`
* `enumerate` sets `s.repos` by iterating over `filteredRepoCache`

At no point is the repo added to the `repoInfoCache`, triggering this case.